lib/libzfs_core/libzfs_core.c

```diff
@@ -176,7 +182,36 @@ libzfs_core_fini(void)
 int
 lzc_ioctl_fd(int fd, unsigned long ioc, zfs_cmd_t *zc)
 {
-	return (lzc_ioctl_fd_os(fd, ioc, zc));
+	if (!g_ioc_trace)
+		return (lzc_ioctl_fd_os(fd, ioc, zc));
+
+	nvlist_t *nvl;
+
+	fprintf(stderr, "=== lzc_ioctl: call: ioc=0x%lx name=%s\n",
+	    ioc, zc->zc_name[0] ? zc->zc_name : "[none]");
+	if (zc->zc_nvlist_src) {
+		nvl = fnvlist_unpack(
+		    (void *)(uintptr_t)zc->zc_nvlist_src,
+		    zc->zc_nvlist_src_size);
+		nvlist_print(stderr, nvl);
+		fnvlist_free(nvl);
+	}
+
+	int rc = lzc_ioctl_fd_os(fd, ioc, zc);
+	int err = errno;
+
+	fprintf(stderr, "=== lzc_ioctl: result: ioc=0x%lx name=%s errno=%d\n",
+	    ioc, zc->zc_name[0] ? zc->zc_name : "[none]", errno);
+	if (!err && zc->zc_nvlist_dst) {
```

Comment:
Is it right to check for `errno` without checking for return value?

Reply:
Good catch! Saving and restoring `errno` was added very late; before that, `rc` used to be called `err`. Fixed, thanks.